qiskit_nature/algorithms/ground_state_solvers/orbital_rotation.py

```diff
@@ -0,0 +1,218 @@
+
+        self._orbital_rotations = []
+        for i in range(half_as):
+            for j in range(half_as):
+                if i < j:
+                    self._orbital_rotations.append([i, j])
+
+    def _create_parameter_list_for_orbital_rotations(self) -> None:
+        """Initializes the initial values of orbital rotation matrix kappa."""
+
+        # creates the indices of matrix kappa and prevent user from trying to rotate only betas
+        if self._orbital_rotations is None:
+            self._create_orbital_rotation_list()
+        elif self._orbital_rotations is None and self._orbital_rotations_beta is not None:
+            raise QiskitNatureError(
+                "Only beta orbitals labels (orbital_rotations_beta) have been provided."
+                "Please also specify the alpha orbitals (orbital_rotations) "
+                "that are rotated as well. Do not specify anything to have by default "
+                "all orbitals rotated."
+            )
+
+        if self._orbital_rotations_beta is not None:
+            num_parameters = len(self._orbital_rotations + self._orbital_rotations_beta)
+        else:
+            num_parameters = len(self._orbital_rotations)
+        self._parameters = [self._parameter_initial_value for _ in range(num_parameters)]
+
+    def _create_parameter_bounds(self) -> None:
+        """Create bounds for parameters."""
+        self._parameter_bounds = [self._parameter_bound_value for _ in range(self._num_parameters)]
+
+    def get_orbital_rotation_matrix(self, parameters: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """Creates 2 matrices K_alpha, K_beta that rotate the orbitals through MO coefficient
+        C_alpha = C_RHF * U_alpha where U = e^(K_alpha), similarly for beta orbitals."""
+
+        self._parameters = parameters  # type: ignore
+        k_matrix_alpha = np.zeros((self._dim_kappa_matrix, self._dim_kappa_matrix))
+        k_matrix_beta = np.zeros((self._dim_kappa_matrix, self._dim_kappa_matrix))
+        # allows to selectively rotate pairs of orbitals
+        if self._orbital_rotations_beta is None:
+            for i, exc in enumerate(self._orbital_rotations):
+                k_matrix_alpha[exc[0]][exc[1]] = self._parameters[i]
+                k_matrix_alpha[exc[1]][exc[0]] = -self._parameters[i]
+                k_matrix_beta[exc[0]][exc[1]] = self._parameters[i]
+                k_matrix_beta[exc[1]][exc[0]] = -self._parameters[i]
+        else:
+            for i, exc in enumerate(self._orbital_rotations):
+                k_matrix_alpha[exc[0]][exc[1]] = self._parameters[i]
+                k_matrix_alpha[exc[1]][exc[0]] = -self._parameters[i]
+
+            for j, exc in enumerate(self._orbital_rotations_beta):
+                k_matrix_beta[exc[0]][exc[1]] = self._parameters[j + len(self._orbital_rotations)]
+                k_matrix_beta[exc[1]][exc[0]] = -self._parameters[j + len(self._orbital_rotations)]
+
```

Comment:
In this moment in Aqua, the freeze core logic is included. We need to figure out a way how to do the
equivalent here (i.e. handle potential transformers being passed to the `problem` instance).
For the time being, as long as your tests do not use any transformers, you can ignore this.